### PR TITLE
Upgrade Storybook to v7.1

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

Upgraded Storybook to v7.1 and fix all the breaking changes.

- Replaced @storybook/react types with the official types.
- Fixed some warnings with named exports.
- Converted stories to CSF 3 syntax.
- Fixed some stories that were not working.
- Migrated from MDX v1 to v2.
- Fixed some MDX files with missing imports.

Issue: XXX-XXXX

## Test plan:

1. Ran `npx storybook@latest upgrade` to upgrade to sb7 and applied all the
migration scripts.

2. Ran `yarn typecheck` to verify that all the types are correct.

3. Ran `yarn start` to verify that the stories are working.

### CODE REVIEW

**Comment by kevinbarabash** [comment]
Is CSF 3 and MDX 2 required to use Storybook 7.x?

File: .storybook/main.js [hunk 1]
```diff
@@ -1,50 +1,33 @@
...
-                    .replace(
-                        wbRegex,
-                        path.resolve(__dirname, "../packages/wonder-blocks$1/src"),
-                    );
-            }),
-        );
-
-        // Return the altered config
-        return config;
-    },
```
Comment (kevinbarabash) [inline .storybook/main.js:49]: I'm happy to see this go away.

**Comment by jandrade** [comment]
> Is CSF 3 and MDX 2 required to use Storybook 7.x?

Not required at this point, but I added them for two reasons:
- CSF3: The change was simple enough with the help of a SB migration tool (codegen).
- MDX2: It will unlock some work that I'm going to be doing around theming.

That said, if we still want to use CSF2/MDX1, it's fine and Storybook won't break if we do it.

File: .github/workflows/node-ci-pr.yml [hunk 2]
```diff
@@ -161,26 +161,3 @@ jobs:
           pattern: "**/dist/es/*.js"
           # Always ignore SourceMaps and node_modules:
           exclude: "{**/*.map,**/node_modules/**}"
-
-  build_storybook:
```
Comment (jandrade) [inline .github/workflows/node-ci-pr.yml:165]: **note:** Found out this step was redundant as this is already done in `chromatic-build.yml`.

File: __docs__/wonder-blocks-birthday-picker/birthday-picker.stories.tsx [hunk 3]
```diff
@@ -32,110 +34,104 @@ export default {
...
+    args: {
+        onChange: () => {},
+        defaultValue: "",
+    },
+    parameters: {
+        chromatic: {
+            disableSnapshot: true,
+        },
     },
 };
```
Comment (kevinbarabash) [inline __docs__/wonder-blocks-birthday-picker/birthday-picker.stories.tsx:55]: Thanks for grouping together like this.

File: .github/workflows/chromatic-build.yml [hunk 4]
```diff
@@ -41,13 +41,20 @@ jobs:
...
+      run: yarn build:storybook
+
     # (if) Run this step on non-dependabot PRs.
     - name: Publish to Chromatic for visual testing (review PRs only)
       if: ${{ inputs.target == 'review' }}
       uses: chromaui/action@v1
       with:
         token: ${{ secrets.GITHUB_TOKEN }}
         projectToken: ${{ secrets.projectToken }}
+        storybookBuildDir: storybook-static
```
Comment (jandrade) [inline .github/workflows/chromatic-build.yml:57]: **note:** This was required by Chromatic due to some internal changes in SB7 (+ the pre-step for building modules + building storybook)

File: .storybook/main.js [hunk 5]
```diff
@@ -1,50 +1,33 @@
...
-        interactionsDebugger: true,
-    },
-    reactOptions: {
-        fastRefresh: true,
-    },
-    staticDirs: ["../static"],
-    // Overriding the default webpack config to allow using storybook with our
-    // monorepo.
-    // See https://storybook.js.org/docs/react/configure/webpack
-    webpackFinal: async (config, {configType}) => {
```
Comment (jandrade) [inline .storybook/main.js:29]: **note:** Thanks to SB7, now we don't need to do this!

File: __docs__/wonder-blocks-banner/banner.stories.tsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import * as React from "react";
 import {StyleSheet} from "aphrodite";
-import type {ComponentStory, ComponentMeta} from "@storybook/react";
+import type {Meta, StoryObj} from "@storybook/react";
```
Comment (jandrade) [inline __docs__/wonder-blocks-banner/banner.stories.tsx:3]: **note:** Most of the changes from here on are around changing the types to v7.

File: package.json [hunk 7]
```diff
@@ -25,7 +25,7 @@
     "lint": "yarn lint:ci .",
     "lint:ci": "eslint --ext .ts --ext .js --ext .tsx --ext .jsx",
     "publish:ci": "yarn run lint:pkg-json && node utils/publish/pre-publish-check-ci.js && git diff --stat --exit-code HEAD && yarn build && yarn build:types && yarn build:flowtypes && changeset publish",
-    "start": "start-storybook -p 6061",
+    "start": "storybook dev -p 6061",
```
Comment (jandrade) [inline package.json:28]: **note:** This is how it works with v7 (same with the build step).

File: static/sb-styles/preview.css
Comment (jandrade) [inline static/sb-styles/preview.css:1]: **note:** Had to change the CSS here due to some SB7 UI revamp.

File: .github/workflows/node-ci-pr.yml [hunk 8]
```diff
@@ -161,26 +161,3 @@ jobs:
...
-        with:
-          node-version: ${{ matrix.node-version }}
-
-      - name: Install & cache node_modules
-        uses: Khan/actions@shared-node-cache-v0
-        with:
-          node-version: ${{ matrix.node-version }}
-
-      - name: Build
-        run: yarn build:storybook
```
Comment (kevinbarabash) [inline .github/workflows/node-ci-pr.yml:186]: We were duplicating effort with this, right?

**Review by kevinbarabash — APPROVED** [review]
LGTM.  I didn't review all of the changes to the individual stories.  If Chromatic is happy with the changes the I'm happy with them as well.  Thanks for updating Storybook.  This unblocks updating TypeScript to 5.x.

File: .github/workflows/node-ci-pr.yml (hunk 8 above)
Comment (jandrade) [reply .github/workflows/node-ci-pr.yml:186]: yep, no idea why/when we added this 🤷

**Pull request merged**